test/e2e/storage/csi_volumes.go

```diff
@@ -282,48 +276,101 @@ var _ = utils.SIGDescribe("CSI Volumes", func() {
 				framework.ExpectNoError(err, "Failed to start pod: %v", err)
 
 				By("Checking if VolumeAttachment was created for the pod")
-				// Check that VolumeAttachment does not exist
 				handle := getVolumeHandle(cs, claim)
 				attachmentHash := sha256.Sum256([]byte(fmt.Sprintf("%s%s%s", handle, scTest.Provisioner, nodeName)))
 				attachmentName := fmt.Sprintf("csi-%x", attachmentHash)
 				_, err = cs.StorageV1beta1().VolumeAttachments().Get(attachmentName, metav1.GetOptions{})
 				if err != nil {
 					if errors.IsNotFound(err) {
-						if test.expectVolumeAttachment {
+						if test.driverAttachable {
 							framework.ExpectNoError(err, "Expected VolumeAttachment but none was found")
 						}
 					} else {
 						framework.ExpectNoError(err, "Failed to find VolumeAttachment")
 					}
 				}
-				if !test.expectVolumeAttachment {
+				if !test.driverAttachable {
 					Expect(err).To(HaveOccurred(), "Unexpected VolumeAttachment found")
 				}
 			})
 		}
 	})
 
-	Context("CSI workload information [Feature:CSIDriverRegistry]", func() {
+	Context("CSI attach test using HostPath driver [Feature:CSIDriverRegistry]", func() {
```

Comment:
@jsafrane do we need to test with both hostpath and mock drivers, or is one enough?

Reply:
IMO mock is enough.